src/vision/status_quo.md

```diff
@@ -28,6 +28,140 @@ These stories may not be true, but they are not fiction. They are based on real-
+
+<details><summary>Making hard choices from a complex ecosystem from the start</summary>
+
+The problems begin from the very first moment a user starts to try out async Rust. The async Rust support in Rust itself is very basic, consisting only of the core Future mechanism. Everything else -- including the basic async runtimes themselves -- lives in user space. This means that users must make a number of choices rom the very beginning:
+
+* what runtime to use
+    * [Barbara makes their first foray into async](status_quo/barbara_makes_their_first_steps_into_async.md)
+    * [Niklaus wants to share knowledge](status_quo/niklaus_wants_to_share_knowledge.md)
+* what http libraries to use
+    * [Barbara anguishes over http](status_quo/barbara_anguishes_over_http.md)
+* basic helpers and utility crates are hard to find, and there are many choices, often with subtle differences between them
+    * [Barbara needs async helpers](status_quo/barbara_needs_async_helpers.md)
+* Furthermore, the async ecosystem is fractured. Choosing one library may entail choosing a specific runtime. Sometimes you may wind up with multiple runtimes running at once.
+    * [Alan started trusting the rust compiler but then async](status_quo/alan_started_trusting_the_rust_compiler_but_then_async.md)
+    * [Barbara needs async helpers](status_quo/barbara_needs_async_helpers.md)
+    * 🚧 Need a story about multiple runtimes working together
+* There is a lack of common, standardized abstractions, which means that often there are multiple attempts to establish common traits and different libraries will employ a distinct subset.
+    * [`Sink` is not implemented by async-std websockets](status_quo/alan_tries_a_socket_sink.md) 
+    * 🚧 No standardized lower-level traits for read, write, iterators in an async setting
+    * 🚧 Lack of widely used higher-level abstractions (like those tower aims to provide)
+    * 🚧 Tokio doesn't support the futures `Stream` trait because of stability concerns
+* Some of the problems are due to the design of Rust itself. The coherence rules in particular.
+    * 🚧 Write about how coherence makes it impossible to establish 
+
+</details>
+
+<details><summary>Once your basic setup is done, the best design patterns are subtle and not always known.</summary>
+
+Writing async programs turns out to have all kinds of subtle tradeoffs. Rust aims to be a language that gives its users control, but that also means that users wind up having to make a lot of choices, and we don't give them much guidance.
+
+* If you need synchronization, you might want an async lock, but you might want a synchronous lock, it's hard to know.
+    * [Alan thinks he needs async locks](status_quo/alan_thinks_he_needs_async_locks.md)
+* Mixing sync and async code is tricky and it's not always obvious how to do it -- something it's not even clear what is "sync" (how long does a loop have to run before you can consider it blocking?)
```

Comment:
The lock story above also touches on this since you may want to opt for a sync lock if the locking is short enough to not really be blocking.